src/main/java/net/dv8tion/jda/api/events/thread/ThreadRevealedEvent.java

```diff
@@ -21,7 +21,13 @@
 
 import javax.annotation.Nonnull;
 
-//TODO-v5: Docs
+/**
+ * This event is dispatched when a {@link ThreadChannel} that JDA didn't previously have access to (due to permissions) is now visible.
+ *
+ * For example, if the bot is given the {@link net.dv8tion.jda.api.Permission.ADMINISTRATOR} permission, any thread channels that the bot could not previously see would be "revealed".
+ *
+ * @see ThreadHiddenEvent
+ */
```

Comment:
We need to double check and see if these events only fire if you were already in the channel or not. We need to document whether you need to be in the channel already or not to receive these events.